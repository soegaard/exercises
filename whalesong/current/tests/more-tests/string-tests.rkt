#lang planet dyoo/whalesong/base

(substring "hello world" 0)
(substring "hello world" 1)
(substring "hello world" 2)
(substring "hello world" 3)
(substring "hello world" 0)
(substring "hello world" 1 1)
(substring "hello world" 1 2)
(substring "hello world" 1 3)
(substring "hello world" 2 2)
(substring "hello world" 2 3)
(substring "hello world" 3 3)
(substring "hello world" 3 4)
(substring "hello world" 3 5)


(list->string '())
(list->string '(#\h #\e #\l #\l #\o))


(string)
(string #\a #\b #\c)

(string-ref "hello" 0)
(string-ref "hello" 1)
(string-ref "hello" 2)
(string-ref "hello" 4)

(string->list "xyz")
(string->list "x")
(string->list "")

(string<? "" "")
(string<? "hello" "Hello")
(string<? "hello" "hello")
(string<? "hello" "hi")
(string<? "hello" "he")
(string<? "a" "a")
(string<? "a" "b")
(string<? "b" "a")
(string<? "b" "b")
(string<? "ba" "ab")
(string<? "ba" "baa")
(string<? "baaa" "baa")

(string<=? "" "")
(string<=? "hello" "Hello")
(string<=? "hello" "hello")
(string<=? "hello" "hi")
(string<=? "hello" "he")
(string<=? "a" "a")
(string<=? "a" "b")
(string<=? "b" "a")
(string<=? "b" "b")
(string<=? "ba" "ab")
(string<=? "ba" "baa")
(string<=? "baaa" "baa")

(string>? "" "")
(string>? "hello" "Hello")
(string>? "hello" "hello")
(string>? "hello" "hi")
(string>? "hello" "he")
(string>? "a" "a")
(string>? "a" "b")
(string>? "b" "a")
(string>? "b" "b")
(string>? "ba" "ab")
(string>? "ba" "baa")
(string>? "baaa" "baa")

(string>=? "" "")
(string>=? "hello" "Hello")
(string>=? "hello" "hello")
(string>=? "hello" "hi")
(string>=? "hello" "he")
(string>=? "a" "a")
(string>=? "a" "b")
(string>=? "b" "a")
(string>=? "b" "b")
(string>=? "ba" "ab")
(string>=? "ba" "baa")
(string>=? "baaa" "baa")

(string=? "" "")
(string=? "hello" "Hello")
(string=? "hello" "hello")
(string=? "hello" "hi")
(string=? "hello" "he")
(string=? "a" "a")
(string=? "a" "b")
(string=? "b" "a")
(string=? "b" "b")
(string=? "ba" "ab")
(string=? "ba" "baa")
(string=? "baaa" "baa")








(string-ci<? "" "")
(string-ci<? "hello" "Hello")
(string-ci<? "hello" "hello")
(string-ci<? "hello" "hi")
(string-ci<? "hello" "he")
(string-ci<? "a" "a")
(string-ci<? "a" "b")
(string-ci<? "b" "a")
(string-ci<? "b" "b")
(string-ci<? "ba" "ab")
(string-ci<? "ba" "baa")
(string-ci<? "baaa" "baa")

(string-ci<=? "" "")
(string-ci<=? "hello" "Hello")
(string-ci<=? "hello" "hello")
(string-ci<=? "hello" "hi")
(string-ci<=? "hello" "he")
(string-ci<=? "a" "a")
(string-ci<=? "a" "b")
(string-ci<=? "b" "a")
(string-ci<=? "b" "b")
(string-ci<=? "ba" "ab")
(string-ci<=? "ba" "baa")
(string-ci<=? "baaa" "baa")

(string-ci>? "" "")
(string-ci>? "hello" "Hello")
(string-ci>? "hello" "hello")
(string-ci>? "hello" "hi")
(string-ci>? "hello" "he")
(string-ci>? "a" "a")
(string-ci>? "a" "b")
(string-ci>? "b" "a")
(string-ci>? "b" "b")
(string-ci>? "ba" "ab")
(string-ci>? "ba" "baa")
(string-ci>? "baaa" "baa")

(string-ci>=? "" "")
(string-ci>=? "hello" "Hello")
(string-ci>=? "hello" "hello")
(string-ci>=? "hello" "hi")
(string-ci>=? "hello" "he")
(string-ci>=? "a" "a")
(string-ci>=? "a" "b")
(string-ci>=? "b" "a")
(string-ci>=? "b" "b")
(string-ci>=? "ba" "ab")
(string-ci>=? "ba" "baa")
(string-ci>=? "baaa" "baa")

(string-ci=? "" "")
(string-ci=? "hello" "Hello")
(string-ci=? "hello" "hello")
(string-ci=? "hello" "hi")
(string-ci=? "hello" "he")
(string-ci=? "a" "a")
(string-ci=? "a" "b")
(string-ci=? "b" "a")
(string-ci=? "b" "b")
(string-ci=? "ba" "ab")
(string-ci=? "ba" "baa")
(string-ci=? "baaa" "baa")

