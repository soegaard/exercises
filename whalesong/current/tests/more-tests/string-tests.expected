"hello world"
"ello world"
"llo world"
"lo world"
"hello world"
""
"e"
"el"
""
"l"
""
"l"
"lo"
""
"hello"
""
"abc"
#\h
#\e
#\l
#\o
(#\x #\y #\z)
(#\x)
()
false
false
false
true
false
false
true
false
false
false
true
false
true
false
true
true
false
true
true
false
true
false
true
false
false
true
false
false
true
false
false
true
false
true
false
true
true
true
true
false
true
true
false
true
true
true
false
true
true
false
true
false
false
true
false
false
true
false
false
false
false
false
false
true
false
false
true
false
false
false
true
false
true
true
true
true
false
true
true
false
true
false
true
false
false
false
false
false
true
false
false
true
false
true
false
true
true
true
true
false
true
true
false
true
true
true
false
true
true
true
true
false
false
true
false
false
true
false
false
false
